### DISCUSSION TITLE
Arm support

### DISCUSSION THREAD

**Discussion by sackley-mccr** [nfrastack/container-mariadb#13] [Ideas]
I'm not super familiar with GitHub commenting so I hope this comment reaches you. I apologize in advance, is there an easy way to  reconfigure this image for arm support? Would a reconfiguration only be possible on your end? i.e Is it possible for me to change anything on my end so you don't have to do anything except reply to this comment?

Thanks in advance,

**Comment by tiredofit** [comment]
If you wanted to build a copy you could build it on an arm system and no changes would need to be made. I keep armv6,arv7, and aarch64 base images of `tiredofit/alpine`. I just haven't built the mariadb variants as the compilation time is quite long. I can look into it.

**Reply by sackley-mccr** [reply]
Alright, I'm a newbie at using Docker much less developing with it, so can you give me a better idea of what you mean by that first part? 

I definitely did not expect you to reply so fast especially on Christmas Eve so thank you for that.

**Reply by tiredofit** [reply]
No worries. Got an outage on one of my clients systems right now so I'm killing time waiting for things to start working again.

You can clone this git image and with Docker installed on your system could build it manually with `docker build`. After that it will be local on your system and usable.  You don't need to edit any of the code or anything, its ready to be built and used.

**Reply by sackley-mccr** [reply]
Oh okay gotcha, makes sense. Thanks.

**Reply by tiredofit** [reply]
I just made this commit https://github.com/tiredofit/docker-mariadb/commit/b16282d8e0fddc49305fc1fec65c84188733b527 which theoretically should enable ARM v7 and ARM v8 builds. If you watch the "Actions" tab you'll see if it passed or failed.

If it passes, you should be able to pull the image on your system normally. If it fails, it may need building on your end. I sadly don't have a arm system other than an old raspberry pi 2 and looking forward in 2023 to getting something with a bit more juice. Let's see if it works out, I am about to sign off for the next few days.

Thanks, btw.

**Reply by sackley-mccr** [reply]
Hey, I built a copy of this repository as you described and it installed just fine but now that the container is running the logs show 
/etc/cont-init.d/10-mariadb: line 165, 178, 188 and line 201 : no such file or directory
chown: No such file or directory 
then it tries restarting 10-mariadb service 50 times and eventually stops trying. 

should I use a different version of the db? not 10? 
Let me know if I can give you more helpful info and thanks in advance

**Reply by tiredofit** [reply]
Interesting. Never seen anything like this before- could be an ARM related issue - Those file issues that you are seeing are the initialization script that is being temporarily created and executed in order to setup a database for the first time.

The line in question where the temp file is created in 162 and it is a standard way of doing things. I wonder however, are we on a read only file system? How are you launching this container? Any arguments, or volume mounts would be useful to know.

**Reply by sackley-mccr** [reply]
I just built the copy as you described and launched the container as root, I didn't add any arguments or specific mount points. I just have it loaded into the base system storage for now. I see what you're saying about line 162. I forgot to mention in my previous message, which now I realize is probably the most important part, the first error I got was "mktemp (null): no such file or directory." 
So INIT_FILE will never work because it referenced mktemp which is already broken.

**Reply by tiredofit** [reply]
Can you try launching this with the docker-compose.yml example that I have in the examples section? The only thing you will need to replace is the "image: " declaration with the name of your own image, and then the proper volumes and mount points are setup here. I'm at a loss as to whats happening here, this definitely shouldn't be popping up just from building from a different platform. I have initialized this image a couple thousand times in production on various archictectures.. It may uncover something that is set with the version of Docker or something else.

We used to have problems with Windows Services for Linux doing funky things to bind mounts - this may be one of them.

**Comment by sackley-mccr** [comment]
Hey yeah, that's how I had already been launching the freescout service. I built my own copy of mariadb and referenced it instead of yours in that docker-compose file. I didn't change any mount points or anything and I just compared your default yml to my production version and nothing seems out of place...

**Reply by tiredofit** [reply]
Ok I am beginning to follow you now, alright. 

So- what kind of system is this ?

**Reply by sackley-mccr** [reply]
Raspberry Pi 4b 8gb ram , with an m.2 drive.

**Reply by tiredofit** [reply]
OK. This should be easy. This has now moved into a technical thing which I'm not the greatest at. 

Are you running 32bit or 64bit? `uname -m` should give some details. I'm working blind here as its been some time that I've worked with arm, I have a few RPI2's and and zeros's doing various things in the house but also still running mariadb (32bit) but not these types of issues. I'm running some old version of debian, and some on an old version of Alpine. If you are really just wanting to get something working for my tiredofit/freescout image you may want to find something that there are people who have access to same architectures. Official mariadb image should work? All my image does is simply add some opinionated defaults for memory management, some monitoring for my clients, and allows for multiple databases to be created - Again mostly based on my experience and my opinions - however a mariadb/ or mysql database is bit by bit the same thing on a filesystem.

The fact that tmp is giving you issues makes me say why dont you mount /tmp to an external volume, however I use /tmp in my containers for state control (should you reboot your host system and the container still persists to run - deviating from the _original_ intent of Docker as I run multiple processes/aka similar to VMs of 2000s).. This is something bigger that probably without access to hardware to test and see it with my eyes I'm not sure I'm going to provide much insight on.

**Comment by tiredofit** [comment]
`tiredofit/mariadb:10.11` now works for armv8 / arm64 processors...
